### ISSUE TITLE
Animation position shifts by  0.0001 second, regardless of animation step size

### ISSUE THREAD

**Issue by SaffronStreams** [godotengine/godot#34431]
I'm using Godot 3.2.beta3 on Windows 10 Home, this issue was the same for beta2.

When using the AnimationPlayer for 2D animation, using a bonerigged (Polygon2D, Skeleton2D, Bone2D) character, when I try to step between keypositions in my animation using the up and down arrows next to the "animation position" text box, the time shifts by 0.0001 second, regardless of what I'm animating on (0.05 second in the example).

I would think this is a bug, since stepping by so little has sadly rendered the arrows useless to me, and I would think it's the same for other animators.

![animationposition](https://user-images.githubusercontent.com/50481103/71097077-b9e4e280-21af-11ea-8edc-c6166d92352a.png)

**Comment by SaffronStreams** [comment]
(Can regular users add labels to an issue? I seem unable to.)

**Comment by Calinou** [comment]
> (Can regular users add labels to an issue? I seem unable to.)

No, only people with write access to the repository can (unless the issue template has automatically-assigned labels).

**Comment by soundgnome** [comment]
I am running into this as well. It does not happen with 3.1.2.

@Calinou I may be able to generate a PR to address this if that would be useful. I have not contributed to Godot before but it does appear related to the change you cited and I just made a local build which at least partially addresses the issue.

**Comment by Calinou** [comment]
@soundgnome Thanks for looking into it! Please open a pull request :slightly_smiling_face:

**Comment by akien-mga** [comment]
This is fixed in 4.0.

**Comment by soundgnome** [comment]
@akien-mga I can still repro this in 4.0rc1. My PR is so stale I think it's totally reasonable for that to be closed, I'll open a new one when I find time to get back to this. But while I understand there are more pressing issues for 4.0 that deserve more immediate attention I don't think this should be closed.
